src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -362,6 +362,12 @@ internal AD7BoundBreakpoint AddBoundBreakpoint(BoundBreakpoint bp)
                 //check can bind one last time. If the pending breakpoint was deleted before now, we need to clean up gdb side
                 if (CanBind())
                 {
+                    foreach( var boundBp in _boundBreakpoints.Where((b) => b.PendingBreakpoint.BreakpointId == boundBreakpoint.PendingBreakpoint.BreakpointId).ToList())
```

Comment:
Is there any situation where the `Where` clause will evaluate to false?
Did multiple bound breakpoints for a single pending (ex: templates) work before? This seems like it will break it now if it worked before. Seems like we really need to add `Number` to a bound breakpoint if we want it to work.

Reply:
@gregg-miskelly can you see if my attempt at the fix will work?